fog/view/server/src/fog_view_shard_responses_processor.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use crate::error::RouterServerError;
+
+use mc_attest_api::attest;
+use mc_fog_api::{view::MultiViewStoreQueryResponse, view_grpc::FogViewApiClient};
+use mc_fog_uri::FogViewUri;
+use std::{str::FromStr, sync::Arc};
+
+/// The result of processing the MultiViewStoreQueryResponse from each Fog View
+/// Shard.
+pub struct ProcessedShardResponseData {
+    /// gRPC clients for Shards that need to be retried for a successful
+    /// response.
+    pub shard_clients_for_retry: Vec<Arc<FogViewApiClient>>,
+
+    /// Uris for *individual* Fog View Stores that need to be authenticated with
+    /// by the Fog Router.
```

Comment:
Per comment in `fog_view_shard_response_processor.rs` suggest expanding the doc comment on `view_store_uris_for_authentication` to help clarify the spacial dependency. 
I'm also not opposed to *not* adding the extra docs if it feels like it's just my lack of overall understanding.
```suggestion
    /// Uris for *individual* Fog View Stores that need to be authenticated with
    /// by the Fog Router.
    /// Should only have entries if `shard_clients_for_retry` has entries.
```

Reply:
Done.